nvflare/private/fed/tbi.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_constant import FLContextKey, SystemConfigs
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.fl_exception import NotReadyToEndRun
+from nvflare.fuel.utils.config_service import ConfigService
+
+
+class TBI(FLComponent):
+
+    """(TBI) Task Based Interaction is the base class for ServerRunner and ClientRunner that implement details of
+    task based interactions.
+
+    TBI implements common behavior of ServerRunner and ClientRunner.
+    """
+
+    def __init__(self):
+        super().__init__()
```

Comment:
minor thing, since we are not adding more stuff in constructor, we can ignore these lines.

Reply:
We may need to do something in the future.